### ISSUE TITLE
`no-useless-spread`: Check more cases

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1482]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/155/badge.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/155)

*(The badge links to a different issue but the bounty applies to this issue)*

## Fail

```js
const array = [
	...foo.map(/* ... */)
];
```

Other array methods returns a new array, eg: `concat()`, `flatMap()`, `copyWithin()`, etc
Other known arrays, eg: `Array.from()`, `Object.keys()`, ...

## Pass

```js
const array = foo.map(/* ... */);
```

These cases are like "useless array copy"

**Comment by rightaway** [comment]
It can check that spread isn't used when there's only a single property. So `{ ...expression }` and `[ ...expression ]` should both just be `expression`.

**Comment by fisker** [comment]
> It can check that spread isn't used when there's only a single property. 

It can be making a shallow copy.

**Comment by rightaway** [comment]
Could checking this be added as an option? Sometimes spread is used just to combine things, so if you start off with `{ ...a, b: null, c: null }` and then later remove `b` and `c` you're left with `{ ...a }` when you really could replace it with `a`.

If there was this option you could disable the rule in cases when you want a shallow object.

**Comment by fisker** [comment]
@rightaway 
I'm going to skip check spreading single UNKNOWN  object, if you still want it please open a separate issue.
